src/librustc_codegen_llvm/back/write.rs

```diff
@@ -393,9 +395,12 @@ pub struct CodegenContext {
     time_graph: Option<TimeGraph>,
     // The assembler command if no_integrated_as option is enabled, None otherwise
     assembler_cmd: Option<Arc<AssemblerCommand>>,
+    // This field is used to give a lifetime parameter to the struct so that it can implement
+    // the Backend trait.
+    phantom: PhantomData<&'ll ()>
```

Comment:
This seems *ok* but I usually find these sorts of things to indicate some other "mis-alignment". This `'ll` lifetime -- what does it logically represent here? Is there some resource we could take a reference to instead? Should it maybe be moved to the `Backend` trait?

Anyway, not a blocking concern, but I do feel like the "semantic meaning" of this lifetime is unclear to me.